### ISSUE TITLE
Feature Request: Better error reporting for tflite conversion

### ISSUE THREAD

**Issue by smoosh911** [tensorflow/tensorflow#17158]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:MacOS
- **TensorFlow installed from (source or binary)**:source
- **TensorFlow version (use command below)**:1.5.0
- **Python version**: 2.7.10
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: Not using
- **GPU model and memory**: Radeon Pro 455
- **Exact command to reproduce**: In macOS terminal: bazel-bin/tensorflow/contrib/lite/toco/toco \
  --input_file=../adventures-in-ml-code/tensorflow_word2vec/frozen_graph.pb \
  --input_format=TENSORFLOW_GRAPHDEF \
  --output_format=TFLITE \
  --output_file=../adventures-in-ml-code/tensorflow_word2vec/word2vec.lite \
  --inference_type=FLOAT \
  --input_arrays=input \
  --output_arrays=output \
  --input_shapes=1 \
  --output_shapes=1

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

Overview
I have been trying to convert a word2vec custom model into tensorflow lite. I am using the code in the tensorflow documentation that was provided as a simple word2vec example.The conversion process up to the freeze_graph.py works fine and I can use that file to run inference no problem in a python script on my desktop. My problem is when I try to convert to lite. 

Model conversation seems to work
When I run the command I listed above I get a .tflite file. I can even visualize this file using bazel-bin/tensorflow/contrib/lite/tools/visualize /Users/miperry/Documents/adventures-in-ml-code/tensorflow_word2vec/word2vec.tflite /Users/miperry/Documents/adventures-in-ml-code/tensorflow_word2vec/word2vec_model_viz.html. The model looks fine in the visualizer too.

Use on mobile doesn't work
When I implement my tflite model into an iOS app and try to grab a node with 
````
float* out = interpreter->typed_tensor<float>();
````
I sometimes get a valid pointer back but other times I get a null pointer depending on the node. 

After retraining my model with different shapes of inputs and outputs I finally noticed documentation at the end of the page https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/tf_ops_compatibility.md that states some operations that are present but not ready for custom models. I am using a couple of these operations and I'm guessing this is the reason my lite model isn't working. 

My reqeust
Please update the errors available for tflite conversion. I don't understand why it would have been difficult to throw an error on tflite conversion that specifies a node and states that tflite doesn't support its operation. Videos posted by google employees and the introduction to tflite make it sound like this product is ready to make life easier when converting for mobile which is what encouraged me to give it a shot. Now I see that this isn't the case yet. If nothing else I would ask for the sanity of your users that tensorflow documentation more clearly warns about the limitations on mobile. 

Tensorflow is a cool concept and the parts that are ready are awesome. Thanks for the great work!

My graph code:
````
valid_dataset = tf.constant(valid_examples, dtype=tf.int32,name="input")

embeddings = tf.Variable(tf.random_uniform([vocabulary_size, embedding_size], -1.0, 1.0),name="embeddings")

norm = tf.sqrt(tf.reduce_sum(tf.square(embeddings), 1, keep_dims=True),name="norm")
normalized_embeddings = tf.div(embeddings, norm, 'normalized_embeddings')
valid_embeddings = tf.nn.embedding_lookup(normalized_embeddings, valid_dataset,name="valid_embeddings")
similarity = tf.matmul(valid_embeddings, tf.transpose(normalized_embeddings), transpose_b=False,name="similarity")
output = tf.reshape(similarity, [-1], name="output")
````

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

Output from tflite conversion:
````
(adventures-in-ml-code) miperry-macOS:tensorflow miperry$ bazel-bin/tensorflow/contrib/lite/toco/toco   --input_file=../adventures-in-ml-code/tensorflow_word2vec/frozen_graph.pb   --input_format=TENSORFLOW_GRAPHDEF   --output_format=TFLITE   --output_file=../adventures-in-ml-code/tensorflow_word2vec/word2vec.tflite   --inference_type=FLOAT   --input_arrays=input   --output_arrays=output   --input_shapes=1   --output_shapes=1000
2018-02-20 13:06:59.037245: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 14 operators, 21 arrays (0 quantized)
2018-02-20 13:06:59.038319: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 14 operators, 21 arrays (0 quantized)
2018-02-20 13:06:59.072631: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After general graph transformations pass 1: 5 operators, 11 arrays (0 quantized)
2018-02-20 13:06:59.072687: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before dequantization graph transformations: 5 operators, 11 arrays (0 quantized)
2018-02-20 13:06:59.072731: I tensorflow/contrib/lite/toco/allocate_transient_arrays.cc:311] Total transient array allocated size: 2401216 bytes, theoretical optimal value: 240000
````

**Comment by asimshankar** [comment]
@aselle @andrehentz should be able to triage / comment on whether `toco` should catch/report unsupported operations.

**Comment by andrehentz** [comment]
I'm surprised toco isn't printing out whatever unsupported operation it is finding. The following code should have been triggered:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/tflite/export.cc#L304

**Comment by smoosh911** [comment]
I'm guessing that it was failing on the embedding lookup? I noticed in the docs that this operation wasn't yet supported but it was defined or something like that. 

If I tried to use the python converter instead of bazel I would get an error about my output size, saying how it should be size 1 when it found 1000. This error was super confusing to me because I didn't find anything in the documentation that says mobile inference was limited to a single output. Also, the flagship tensorflow lite image object recognition example has many output labels (I think 1000).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by aselle** [comment]
@smoosh911, sorry for the difficulty you are having. We are sorry we weren't clearer that tflite was in developer preview and that means that not-everything is working as well as we would like. I'd like to work to make your case work. Probably the embedding lookup is the difficulty as you say. Could you provide an example of what you are using for vocabulary and valid_dataset?

**Comment by smoosh911** [comment]
If you could get it working on the basic example provided here https://www.tensorflow.org/tutorials/word2vec then it will work for me.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ghtwht** [comment]
Actually, toco supports the embedding_lookup operator, but you should be sure that the input is one dimensional only, or the output dimension will be wrong.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jdduke** [comment]
We've landed quite a few improvements to the converter, including a new conversion backend in the upcoming 2.2 release.
